src/lib/cpd/CpdEngine.ts

```diff
@@ -72,34 +121,47 @@ export class CpdEngine extends AbstractRuleEngine {
 		return results;
 	}
 
-	private sortPaths(targets: RuleTarget[]): Map<LANGUAGE, string[]> {
-		const languageToPaths = new Map();
-
+	//visible to unit tests
+	sortPaths(targets: RuleTarget[]): Map<LANGUAGE, string[]> {
+		const languageToPaths = new Map<LANGUAGE, string[]>();
+		const unmatchedPaths: string[] = [];
 		for (const target of targets) {
 			for (const path of target.paths) {
-				const i = path.lastIndexOf(".");
-				if (i === -1) {
-					uxEvents.emit('warning-always', `Target: '${path}' was not processed by CPD, no file extension found.`);
-					continue;
+				if (!this.matchPathToLanguage(path, languageToPaths)) {
+					// If language could not be identified, note down the path
+					unmatchedPaths.push(path);
 				}
-				const ext = path.substr(i).toLowerCase();
-				if (FILE_EXTS_TO_LANGUAGE.has(ext)) {
-					if (this.validCPDLanguages.includes(FILE_EXTS_TO_LANGUAGE.get(ext))) {
-						const language = FILE_EXTS_TO_LANGUAGE.get(ext);
-						if (languageToPaths.has(language)) {
-							languageToPaths.get(language).push(path);	
-						} else {
-							languageToPaths.set(language, [path]);
-						}
-					} else {
-						uxEvents.emit('warning-always', `Target: '${path}' was not processed by CPD, language '${FILE_EXTS_TO_LANGUAGE.get(ext)}' not supported.`);
-					}	
+			}
+		}
+
+		// Let user know about file paths that could not be matched
+		if (unmatchedPaths.length > 0) {
+			uxEvents.emit('info-verbose', eventMessages.getMessage('info.unmatchedPathExtensionCpd', [unmatchedPaths.join(",")]));
+		}
+
+		return languageToPaths;
+	}
+
+	/**
+	 * Identify the language of a file using the file extension
+	 * @param path to be examined
+	 * @param languageToPaths map with entries of language to paths matched so far
+	 * @returns true if the language was identifed and false if not
+	 */
+	private matchPathToLanguage(path: string, languageToPaths: Map<LANGUAGE, string[]>): boolean {
+		const ext = path.slice(path.lastIndexOf(".") + 1);
```

Comment:
Should this lowercase `ext` in case the user has a file named `foo.Java` or something else unexpected?

Reply:
Missed this. Will fix it.